test/integration/homepage_popular_links_test.rb

```diff
@@ -51,6 +51,48 @@ class HomepagePopularLinksTest < JavascriptIntegrationTest
     end
   end
 
+  context "#confirm_destroy" do
+    setup do
+      click_button("Create new edition")
+    end
+
+    should "show the 'Delete draft' confirmation page with the option to actually delete" do
+      click_link("Delete draft")
+
+      assert page.has_text?("Delete draft")
+      assert page.has_text?("Are you sure you want to delete this draft?")
+      assert page.has_text?("Cancel")
+    end
+
+    should "revert to show page if delete is cancelled" do
+      click_link("Delete draft")
+      click_link("Cancel")
+
+      assert_title "Popular on GOV.UK"
+    end
+  end
+
+  context "#destroy" do
+    setup do
+      click_button("Create new edition")
+    end
+
+    should "show the previously published edition when a draft is deleted" do
```

Comment:
I think you have kind of covered this functionality in controller test

Reply:
Can I just get some clarification on the above? Are you agreeing that the test should stay? Does it need changes?

Reply:
Personally, I'd leave it as-is.